restapi/src/main/java/io/stargate/web/service/WhereParser.java

```diff
@@ -15,168 +15,242 @@
-          Column.ColumnType valueType = mapType.parameters().get(1);
-          Object mapKey = Converters.toCqlValue(keyType, entryKey.asText());
-          Object mapValue = Converters.toCqlValue(valueType, entryValue.asText());
-          conditions.add(
-              BuiltCondition.of(
-                  LHS.mapAccess(fieldName.toLowerCase(), mapKey), op.predicate, mapValue));
-        } else {
-          // Remaining operators: the value is a simple node
-          if (!value.isValueNode()) {
-            throw new IllegalArgumentException(
-                String.format(
-                    "Value entry for field %s, operation %s was expecting a value, but found an object or array.",
-                    fieldName, rawOp));
-          }
-
-          if (op == FilterOp.$EXISTS) {
-            if (!value.isBoolean() || !value.booleanValue()) {
-              throw new IllegalArgumentException("`exists` only supports the value `true`.");
-            }
-            conditions.add(conditionToWhere(fieldName, op, true));
-          } else {
-            Object val = value.asText();
-            Column column = tableData.column(fieldName);
-            if (column == null) {
-              throw new IllegalArgumentException(
-                  String.format("Unknown field name '%s' in where clause.", fieldName));
-            }
-            Column.ColumnType columnType = column.type();
-            if (columnType != null) {
-              Column.ColumnType valueType;
-              if (op == FilterOp.$CONTAINS) {
-                if (columnType.isCollection()) {
-                  valueType =
-                      columnType.isMap()
-                          ? columnType.parameters().get(1)
-                          : columnType.parameters().get(0);
-                } else {
-                  throw new IllegalArgumentException(
-                      String.format(
-                          "Field %s: operation %s is only supported for collection types",
-                          fieldName, rawOp));
-                }
-              } else if (op == FilterOp.$CONTAINSKEY) {
-                if (columnType.isMap()) {
-                  valueType = columnType.parameters().get(0);
-                } else {
-                  throw new IllegalArgumentException(
-                      String.format(
-                          "Field %s: operation %s is only supported for map types",
-                          fieldName, rawOp));
-                }
-              } else {
-                valueType = columnType;
-              }
-              val = Converters.toCqlValue(valueType, value.asText());
+        QueryContext context = new QueryContext();
+        context.fieldName = fieldName;
+        context.rawOp = rawOp;
+        context.operator = operator;
+        context.value = value;
+        context.type = column.type();
+
+        switch (operator) {
+          case $IN:
+            evaluateIN(conditions, context);
```

Comment:
nit: `evaluateIn`

Reply:
Thanks @EricBorczuk for your time to review and your comments ! 🙂